include/filters/filter_base.hpp

```diff
@@ -376,77 +228,47 @@ class MultiChannelFilterBase : public FilterBase<T>
 {
 public:
   MultiChannelFilterBase():number_of_channels_(0){};
+
+  /** \brief Virtual Destructor
+   */
+  virtual ~MultiChannelFilterBase() = default;
+
+  // Allow overloading with base class function
+  using FilterBase<T>::configure;
   
   /** \brief Configure the filter from the parameter server 
    * \param number_of_channels How many parallel channels the filter will process
    * \param The parameter from which to read the configuration
    * \param node_handle The optional node handle, useful if operating in a different namespace.
    */
-  bool configure(unsigned int number_of_channels, const std::string& param_name, ros::NodeHandle node_handle = ros::NodeHandle())
-  {
-    XmlRpc::XmlRpcValue config;
-    if (!node_handle.getParam(param_name, config))
-    {
-      ROS_ERROR("Could not find parameter %s on the server, are you sure that it was pushed up correctly?", param_name.c_str());
-      return false;
-    }
-    return configure(number_of_channels, config);
-    
-  }
-
-
-  /** \brief The public method to configure a filter from XML 
-   * \param number_of_channels How many parallel channels the filter will process
-   * \param config The XmlRpcValue to load the configuration from 
-   */
-  bool configure(unsigned int number_of_channels, XmlRpc::XmlRpcValue& config)
+  bool configure(
+    size_t number_of_channels,
+    const std::string& param_prefix,
+    const std::string& filter_name,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr & node_logger,
+    const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr & node_params)
   {
-    ROS_DEBUG("FilterBase being configured with XmlRpc xml: %s type: %d", config.toXml().c_str(), config.getType());
-    if (configured_)
-    {
-      ROS_WARN("Filter %s of type %s already being reconfigured", filter_name_.c_str(), filter_type_.c_str());
-    };
-    configured_ = false;
     number_of_channels_ = number_of_channels;
-    ROS_DEBUG("MultiChannelFilterBase configured with %d channels", number_of_channels_);
-    bool retval = true;
 
-    retval = retval && FilterBase<T>::loadConfiguration(config);
-    retval = retval && configure();
-    configured_ = retval;
-    return retval;
+    return configure(param_prefix, filter_name, node_logger, node_params);
```

Comment:
I know the `using` line is right above, but I would find this much more readable if we removed the `using` line and just fully-qualified this here:

```
return FilterBase<T>::configure(param_prefix, filter_name, node_logger, node_params);
```

Reply:
`FilterBase<T>` instead of sidestepping name hiding in f8550c74e663b1620137708ec1e8b205ad470b9b